File: Leetcode/746 - Min Cost Climbing Stairs/solution.py
def minCostClimbingStairs(self, cost: List[int]) -> int:
    prev2, prev1 = cost[0], cost[1]

    for i in range(2, len(cost)):
        current = cost[i] + min(prev1, prev2)
        prev2, prev1 = prev1, current

    return min(prev1, prev2)

File: Leetcode/746 - Min Cost Climbing Stairs/solution.js
var minCostClimbingStairs = function(cost) {
    let prev2 = cost[0], prev1 = cost[1];

    for (let i = 2; i < cost.length; i++) {
        let current = cost[i] + Math.min(prev1, prev2);
        prev2 = prev1;
        prev1 = current;
    }

    return Math.min(prev1, prev2);
};